### PR TITLE
Templatizes current version of envoy

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This allows us to cache the download of envoy when it doesn't change.

### CODE REVIEW

File: pkg/cmd/run.go [hunk 1]
```diff
@@ -41,29 +42,25 @@ var (
 // NewRunCmd create a command responsible for starting an Envoy process
 func NewRunCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "run <reference|filepath> [flags] [-- <envoy-args>]",
```
Comment (codefromthecrypt) [inline pkg/cmd/run.go:44]: this didn't work so removed it

File: pkg/reference.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 Tetrate
```
Comment (codefromthecrypt) [inline pkg/reference.go:1]:
```suggestion
// Copyright 2021 Tetrate
```

File: .github/workflows/commit.yaml [hunk 3]
```diff
@@ -121,10 +122,11 @@ jobs:
         uses: actions/cache@v2
         with:
           path: |
+            ~/.getenvoy/builds
```
Comment (codefromthecrypt) [inline .github/workflows/commit.yaml:125]: I need to look into multiple restore keys, if possible because these versions have nothing to do with eachother

Comment (codefromthecrypt) [reply .github/workflows/commit.yaml:129]: documented

**Comment by codefromthecrypt** [comment]
cache works

Screen Shot 2021-04-21 at 1 11 18 PM

File: pkg/binary/envoytest/util.go [hunk 4]
```diff
@@ -27,20 +27,17 @@ import (
...
-// This currently latest, but we may support a range at some point.
-var reference = "standard:1.17.1"
-
 var once sync.Once
 var errorFetchingEnvoy error
 
 // FetchEnvoyAndRun retrieves the Envoy indicated by reference only once. This is intended to be used with TestMain.
-// In CI, you can execute this to obviate latency during test runs: "go run cmd/getenvoy/main.go fetch standard:1.17.1"
+// Execute this to obviate latency during test runs: go run cmd/getenvoy/main.go fetch $(cat pkg/reference.txt)
 func FetchEnvoyAndRun(m *testing.M) {
```
Comment (codefromthecrypt) [inline pkg/binary/envoytest/util.go:41]: a later change will move tests that use this into the e2e thing and make this go away

**Comment by codefromthecrypt** [comment]
looks like total 500MB cached now (dependencies and envoy binary)

**Comment by mathetake** [comment]
so how much this makes tests faster..?

**Comment by codefromthecrypt** [comment]
8s down to 2s, though there's probably some increase in time unpacking the cache.

Faster isn't the only thing here, it is also more reliable and less clutter log output. Ex no 503 on bintray unless the version changes as the cache is local. Even bigger than that is less copy/paste as that did drift in the past

Before:
```
Wed, 21 Apr 2021 22:53:36 GMT
Run go run cmd/getenvoy/main.go fetch standard:1.17.1 

—snip—

Wed, 21 Apr 2021 22:53:44 GMT go test ./pkg/...
```

After:
```
Wed, 21 Apr 2021 14:02:28 GMT
Run go run cmd/getenvoy/main.go fetch $(cat pkg/reference.txt) 

Wed, 21 Apr 2021 14:02:30 GMT standard:1.17.1/linux-glibc is already downloaded 

Wed, 21 Apr 2021 14:02:30 GMT go test ./pkg/...
```

**Comment by codefromthecrypt** [comment]
IOTW, test execution wouldn't be faster unless we had a drift bug somewhere because we pre-seeded the cache before and after.

**Review by mathetake — APPROVED** [review]
Thanks!

**Comment by codefromthecrypt** [comment]
Thanks again for asking why @mathetake

**Pull request merged**